hw/scripts/pyocd.sh

```diff
@@ -26,6 +26,18 @@ CONNECTION_MODE_DEFAULT="halt"
 # TARGET must be correectly set by the BSP; check `pyocd list --targets`
 # CONNECTION_MODE can be one of "attach", "halt" (default), "pre-reset", "under-reset"
 #
+pyocd_sn() {
+  if [ -n "$JLINK_SN" ]; then
+      UID_ARG="-u $JLINK_SN"
+  elif [ -n "$STLINK_SN" ]; then
+      UID_ARG="-u $STLINK_SN"
+  elif [ -n "$CMSIS_SN" ]; then
+      UID_ARG="-u $CMSIS_SN"
+  else
+      UID_ARG=""
+  fi
```

Comment:
In other places indent is consistently set to 4, here is partial 2 and partial 4